src/kernel_abi.h

```diff
@@ -497,9 +497,18 @@ struct BaseArch : public wordsize, public FcntlConstants {
   struct shmid64_ds {
     ipc64_perm shm_perm;
     size_t shm_segsz;
-    uint64_t shm_atime_only_little_endian;
-    uint64_t shm_dtime_only_little_endian;
-    uint64_t shm_ctime_only_little_endian;
+    __kernel_time_t shm_atime;
+#if __BITS_PER_LONG != 64
+    unsigned long unused1;
+#endif
```

Comment:
I don't understand.  The issue here is structure packing.  The sizes of the variables themselves are already correct on all three architectures, but on x86-32 a uint64_t can be 32 bit aligned, while on ARM 64 bit values require 8 byte alignment.  Thus declaring it with 64 bit values in rr while the kernel uses two 32 bit values leads to incorrect structure packing.